_test/test_sqw/test_equal_to_tol.m

```diff
@@ -0,0 +1,227 @@
+    end
+
+    function test_paged_sqw_objects_equal_if_pix_within_each_bin_shuffled(obj)
+        original_sqw = copy(obj.sqw_2d_paged);
+        npix = [10, 5, 6, 3, 6];
+
+        data = rand(PixelData.DEFAULT_NUM_PIX_FIELDS, 30);
+        shuffled_data = obj.shuffle_pixel_bin_rows(PixelData(data), npix).data;
+
+        npix_in_page = 11;
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+        shuffled_pix = obj.get_pix_with_fake_faccess(shuffled_data, npix_in_page);
+
+        % Replace sqw objects' npix and pix arrays
+        original_sqw.data.npix = npix;
+        original_sqw.data.pix = pix;
+        shuffled_sqw = copy(original_sqw);
+        shuffled_sqw.data.pix = shuffled_pix;
+
+        assertTrue(equal_to_tol(shuffled_sqw, original_sqw));
+    end
+
+    function test_paged_sqw_ne_if_pix_within_bin_shuffled_and_reorder_false(obj)
+        original_sqw = copy(obj.sqw_2d_paged);
+        npix = [10, 5, 6, 3, 6];
+
+        data = rand(PixelData.DEFAULT_NUM_PIX_FIELDS, 30);
+        shuffled_data = obj.shuffle_pixel_bin_rows(PixelData(data), npix).data;
+
+        npix_in_page = 11;
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+        shuffled_pix = obj.get_pix_with_fake_faccess(shuffled_data, npix_in_page);
+
+        % Replace sqw objects' npix and pix arrays
+        original_sqw.data.npix = npix;
+        original_sqw.data.pix = pix;
+        shuffled_sqw = copy(original_sqw);
+        shuffled_sqw.data.pix = shuffled_pix;
+
+        assertFalse(equal_to_tol(shuffled_sqw, original_sqw, 'reorder', false));
+    end
+
+    function test_the_same_sqw_objects_are_equal_with_paged_pix(obj)
+        sqw_copy = obj.sqw_2d_paged;
+        assertTrue(equal_to_tol(obj.sqw_2d_paged, sqw_copy));
+    end
+
+    function test_the_same_sqw_objs_eq_if_fraction_of_pix_and_pix_paged(obj)
+        sqw_copy = obj.sqw_2d_paged;
+        assertTrue(equal_to_tol(obj.sqw_2d_paged, sqw_copy, 'fraction', 0.5));
+    end
+
+    function test_paged_and_non_paged_version_of_same_sqw_file_are_equal(obj)
+        assertTrue(equal_to_tol(obj.sqw_2d_paged, obj.sqw_2d, 'fraction', 0.5));
+    end
+
+    function test_false_returned_if_NaNs_in_sqw_and_nan_equal_is_false(obj)
+        original_sqw = copy(obj.sqw_2d);
+        original_sqw.data.s(5) = nan;
+        sqw_copy = copy(original_sqw);
+
+        assertFalse(equal_to_tol(original_sqw, obj.sqw_2d, 'nan_equal', false));
+        assertFalse(equal_to_tol(sqw_copy, original_sqw, 'nan_equal', false));
+    end
+
+    function test_true_returned_if_NaNs_in_sqw_and_nan_equal_is_true(obj)
+        original_sqw = copy(obj.sqw_2d);
+        original_sqw.data.s(5) = nan;
+        sqw_copy = copy(original_sqw);
+
+        assertFalse(equal_to_tol(original_sqw, obj.sqw_2d, 'nan_equal', true));
+        assertTrue(equal_to_tol(sqw_copy, original_sqw, 'nan_equal', true));
+    end
+
+    function test_using_fraction_argument_only_tests_fraction_of_the_bins(obj)
```

Comment:
this test would benefit from some additional comments; it's not obvious that you're building a test data with a pattern of equal and unequal pixels, e.g. 1,1,1,10,1,1,1,10 in order that the fraction will only compare the equal pixels